client/src/components/Field.jsx

```diff
@@ -17,38 +18,68 @@ const fieldData = {
 };
 
 const Field = ({ typeName, fieldName, returnType, updateEdge, relationship }) => {
+  const nodes = useNodes();
+  const updateNodeInternals = useUpdateNodeInternals();
+  const [handlePosition, setHandlePosition] = useState('right');
+  // const [targetNode, setTargetNode] = useState(null);
+  // const [currNode, setCurrNode] = useState(null);
 
   useEffect(() => {
     // for our vSchema:
     // relationship is a key on a field object that only exists if that field points to a type
     // its value corresponds 1:1 to the object type name and its node's id
     if (relationship) {
+
       const targetType = relationship;
+
       updateEdge({
         id: `${typeName}/${fieldName}-${targetType}`,
         source: typeName,
         sourceHandle: `${typeName}/${fieldName}`,
         target: targetType,
+        // smooth: true,
+        // type: 'smart',
         markerEnd: {
           type: MarkerType.ArrowClosed,
           color: 'cornflowerblue',
           width: 20,
           height: 20,
           strokeWidth: .3
         },
-        animated: true,
+        // animated: true,
         style: { stroke: 'cornflowerblue' },
       });
     }
   }, []);
 
+  // I originally tried to store curr and target nodes in state
+  // and assign them only once in useEffect, however ... that created
+  // all sorts of unintended behavior
+  // You'd think it'd be easier that way ... it seems 'references' got lost in state
+  // So here, we're 'brute forcing' instead.
+  if (relationship) {
+    const targetNode = nodes.find(node => node.id === relationship);
+    const currNode = nodes.find(node => node.id === typeName);
+    const targetPosition = targetNode.position;
+    const currPosition = currNode.position;
+    if (currPosition.x > targetPosition.x && handlePosition !== 'left') {
```

Comment:
one note / question without seeing it: if edges properly avoid nodes, do we need to factor some sort of "gapSize" into this check to let the edge through before the switch?  

Maybe silly thought - just popped in my head

Reply:
Atm, edges are not avoiding nodes -- that is a complex feature I worked on but realized it was going to take too much, so I'm waiting to come back to it later :)